### PR TITLE
Enable RowIter2 on IndexedTableAccess

### PR DESCRIPTION
**Pull request by zachmu**
This is pretty limited and only works for static lookups, not joins, but is enough to benchmark with

This change also does some refactoring of index analysis code, but no other functional changes

Small bug fix to the String() method of SubstringIndex

### CODE REVIEW

File: sql/analyzer/indexes.go [hunk 1]
```diff
@@ -341,53 +285,67 @@ func getComparisonIndexLookup(
...
 	}
 
-	if !isEvaluable(left) && isEvaluable(right) {
-		gf := expression.ExtractGetField(left)
-		if gf == nil {
-			return nil, nil
-		}
+	if isEvaluable(left) || !isEvaluable(right) {
+		return nil, nil
+	}
```
Comment (andrew-wm-arthur) [inline sql/analyzer/indexes.go:290]: doing gods work here, we have way too many giga-blocks

File: sql/expression/function/substring.go [hunk 2]
```diff
@@ -277,7 +277,7 @@ func (s *SubstringIndex) IsNullable() bool {
 }
 
 func (s *SubstringIndex) String() string {
-	return fmt.Sprintf("SUBSTRING_INDEX(%s, %s, %d)", s.str, s.delim, s.count)
+	return fmt.Sprintf("SUBSTRING_INDEX(%s, %s, %s)", s.str, s.delim, s.count)
```
Comment (andrew-wm-arthur) [inline sql/expression/function/substring.go:280]: seems wrong

**Review by andrew-wm-arthur — APPROVED** [review]
LGTM

File: sql/expression/function/substring.go (hunk 2 above)
Comment (zachmu) [reply sql/expression/function/substring.go:280]: s.count is of type Expression (usually a Literal)

Comment (zachmu) [reply sql/expression/function/substring.go:280]: %d was doing something really weird with it, not sure how it ever worked

File: sql/analyzer/indexes.go (hunk 1 above)
Comment (zachmu) [reply sql/analyzer/indexes.go:290]: 💪

**Pull request merged**